server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -438,16 +477,34 @@ public static State fromValue(byte value) {
 
     private final List<Entry> entries;
 
+    private static boolean assertConsistentEntries(List<Entry> entries) {
+        final Map<String, Set<ShardId>> assignedShardsByRepo = new HashMap<>();
+        for (Entry entry : entries) {
+            for (ObjectObjectCursor<ShardId, ShardSnapshotStatus> shard : entry.shards()) {
+                if (shard.value.isActive()) {
+                    assert assignedShardsByRepo.computeIfAbsent(entry.repository(), k -> new HashSet<>()).add(shard.key) :
+                            "Found duplicate shard assignments in " + entries;
+                }
+            }
+        }
+        return true;
+    }
+
     public static SnapshotsInProgress of(List<Entry> entries) {
         if (entries.isEmpty()) {
             return EMPTY;
         }
         return new SnapshotsInProgress(Collections.unmodifiableList(entries));
     }
 
-    private SnapshotsInProgress(List<Entry> entries) {
-        this.entries = entries;
-    }
+    private SnapshotsInProgress(List <Entry> entries) {
+            this.entries = entries;
+            try {
+                assert assertConsistentEntries(entries);
+            } catch (AssertionError e) {
+                throw e;
```

Comment:
I'm not sure to understand why we catch and rethrow here

Reply:
So I could put a debug breakpoint there :D  Thanks for spotting!